tests/integration-tests/tests/performance_tests/test_startup_time.py

```diff
@@ -0,0 +1,126 @@
+
+
+def evaluate_data(value, data):
+    standard_deviation = statistics.stdev(data)
+    mean = statistics.mean(data)
+    logging.info(f"Mean: {mean}")
+    logging.info(f"Standard deviation: {standard_deviation}")
+
+    distance = abs(mean - value) / standard_deviation
+    if value < (mean + 2 * standard_deviation) or value > (mean - 2 * standard_deviation):
+        return False, distance
+    return True, distance
+
+
+def get_data(instance_type, os, cw_client):
+    data = []
+    dimensions = [{"Name": "InstanceType", "Value": instance_type}, {"Name": "OS", "Value": os}]
+
+    cluster_metrics = cw_client.list_metrics(
+        Namespace="ParallelCluster", MetricName="StartupTime", Dimensions=dimensions
+    )
+
+    logging.info(f"Metrics: {cluster_metrics}")
+
+    for metric in cluster_metrics["Metrics"]:
+        result = cw_client.get_metric_statistics(
+            Namespace="ParallelCluster",
+            MetricName="StartupTime",
+            Dimensions=metric["Dimensions"],
+            StartTime=datetime.now() - relativedelta(years=1),
+            EndTime=datetime.now(),
+            Period=30000,
+            Statistics=["Average"],
+            Unit="None",
+        )
+
+        logging.info(f"Results: {result}")
+
+        if result["Datapoints"]:
+            value = result["Datapoints"][0].get("Average")
+            data.append(value)
+    return data
+
+
+def get_metric(os, cluster, instance_type, instance_id, cw_client):
+    value = None
+
+    dimensions = [
+        {"Name": "OS", "Value": os},
+        {"Name": "InstanceID", "Value": instance_id},
+        {"Name": "ClusterName", "Value": cluster.name},
+        {"Name": "InstanceType", "Value": instance_type},
+    ]
+
+    result = cw_client.get_metric_statistics(
+        Namespace="ParallelCluster",
+        MetricName="StartupTime",
+        Dimensions=dimensions,
+        StartTime=datetime.now() - relativedelta(years=1),
+        EndTime=datetime.now(),
+        Period=30000,
+        Statistics=["Average"],
+        Unit="None",
+    )
+
+    logging.info(f"Results: {result}")
+
+    if result["Datapoints"]:
+        value = result["Datapoints"][0].get("Average")
+    return value
+
+
+def test_startup_time(pcluster_config_reader, clusters_factory, test_datadir, region, instance, os, scheduler):
+    cluster_config = pcluster_config_reader()
+    cluster = clusters_factory(cluster_config)
+
+    cw_client = boto3.client("cloudwatch", region_name=region)
+
+    instances = describe_cluster_instances(
+        cluster.name,
+        region,
+        filter_by_node_type="Compute",
+    )
+
+    performance_degradation = {}
+
+    for instance in instances:
+        instance_type = instance["InstanceType"]
+        instance_id = instance["InstanceId"]
+        logging.info(f"Type: {instance_type}")
+
+        # get new startup time
+
+        value = get_metric(os, cluster, instance_type, instance_id, cw_client)
+        logging.info(f"Value: {value}")
```

Comment:
I suggest to be more specific on this log line and say "Observed Startup Time for instance ${instance_id} (${instance_type}) of cluster ${cluster_name}: ${value} seconds" (os can be avoided in the log line because it is reported in the test name that is the prefix of every log line). I assumed it is in seconds, if not let's use the appropriate unit of measure.